pages/api_integration-full-node-streaming.md

```diff
@@ -0,0 +1,437 @@
+
+
+## Enabling Streaming
+The orderbook stream is implemented with gRPC, a streaming RPC protocol developed by Google and used in CosmosSDK. Use the following flags to configure the gRPC streaming feature:
+
+| CLI Flag | Type | Default | Short Explanation |
+| -------- | ----- | ------- | -------- |
+| grpc-streaming-enabled | bool | false | Toggle on to enable full node streaming. Can only be used when grpc is enabled. |
+| grpc-streaming-flush-interval-ms | int | 50 | Buffer flush interval for batch emission of protocol-side updates |
+| grpc-streaming-max-batch-size | int | 2000 | Maximum protocol-side update buffer before dropping all streaming connections|
+| grpc-streaming-max-channel-buffer-size | int | 2000 | Maximum channel size before dropping slow or erroring grpc connections. Decreasing this will more aggressively drop slow client connections. |
+
+**Disclaimer:** We recommend you use this exclusively with your own node, as supporting multiple public gRPC streams with unknown client subscriptions may result in degraded performance.
+
+## Connecting to the Stream
+
+After setting up a full node with gRPC streaming enabled, you can connect to the stream using any gRPC client.
+To follow along with [Google's documentation on gRPC streaming clients](https://grpc.io/docs/languages/go/basics/#client):
+
+1. Clone the [github.com/dydxprotocol/v4-chain](https://github.com/dydxprotocol/v4-chain) repository at the same version as your full node.
+2. Generate the protos: `make proto-gen && make proto-export-deps`.
+3. The generated protos are now in the `.proto-export-deps` directory.
+4. Use the protobuf compiler (protoc) to [generate stubs](https://protobuf.dev/getting-started/) in any supported language.
+5. Follow the documentation to write a streaming client.
+6. Connect to the stream defined in the `dydxprotocol.clob.Query` service ([StreamOrderbookUpdates](https://github.com/dydxprotocol/v4-chain/blob/4199c3e7b00ded24774d49ce8adcbaaa8325ddc1/proto/dydxprotocol/clob/query.proto#L63-L67)).
+
+For Python, the corresponding code is already generated in [the v4-proto PyPi package](https://pypi.org/project/v4-proto/).
+
+## Maintaining Orderbook State 
+
+### Overview
+
+1. Connect to the stream and subscribe to updates for a series of clob pair ids, each of which corresponds to a tradeable instrument.
+2. Discard messages until you receive a `StreamOrderbookUpdate` with `snapshot` set to `true`. This message contains the full orderbook state for each clob pair.
+3. When you see an `OrderPlaceV1` message, insert the order into the book at the end of the queue on its price level. Track the order's initial quantums (quantity) and total filled quantums.
+4. When you see an `OrderUpdateV1` message, update the order's total filled quantums.
+5. When you see a `ClobMatch` (trade) message, update the total filled quantums for each maker order filled using the `fill_amounts` field. 
```

Comment:
I feel like adding some note as to why there's both `OrderUpdateV1` and `ClobMatch` would be helpful. I could imagine this be really confusing why there are two different messages that feel like they are keeping tracking of the exact same thing.